float64 desiredDir
float64 distToGoal
bool gotDir
bool gotDist
